packages/gatsby-remark-code-repls/src/gatsby-node.js

```diff
@@ -36,45 +36,48 @@ exports.createPages = (
     )
   }
 
-  // TODO We could refactor this to use 'recursive-readdir' instead,
-  // And wrap with Promise.all() to execute createPage() in parallel.
-  // I'd need to find a way to reliably test error handling though.
-  const files = recursiveReaddir(directory)
+  try {
+    const files = await readdir(directory)
+    if (files.length === 0) {
+      console.warn(`Specified REPL directory "${directory}" contains no files`)
 
-  if (files.length === 0) {
-    console.warn(`Specified REPL directory "${directory}" contains no files`)
-
-    return
-  }
+      return
+    }
 
-  files.forEach(file => {
-    if (extname(file) === `.js` || extname(file) === `.jsx`) {
-      const slug = file
-        .substring(0, file.length - extname(file).length)
-        .replace(new RegExp(`^${directory}`), `redirect-to-codepen/`)
-      const code = fs.readFileSync(file, `utf8`)
+    Promise.all(
```

Comment:
This is still synchronous right? Why the change?

Reply:
@DSchau it was an attempt at this refactoring:

https://github.com/gatsbyjs/gatsby/blob/9bd0a276d4de922541651d7111b55be7d52f5e4d/packages/gatsby-remark-code-repls/src/gatsby-node.js#L39-L42

However, as you pointed at it’s still synchronous, because `recursive-readdir` returns all the files at once, instead of one by one. Updated the PR to drop the refactoring part, what do you think?